### DISCUSSION TITLE
Supporting several sequences

### DISCUSSION THREAD

**Discussion by GreenRuslan** [godbout/kindaVim.blahblah#197] [Ideas]
The motivation for this request is the fact that I use kj sequence as an escape in all my editors. But the main problem is that I use several languages to communicate. Navigation works fine with another language, but not the sequence to return to the normal mode.

**Comment by godbout** [comment]
sorry, i don't understand the issue you're having.

> The motivation for this request is the fact that I use kj sequence as an escape in all my editors.

ok till now it's fine. you can use `kj` in kV.

> But the main problem is that I use several languages to communicate. Navigation works fine with another language, but not the sequence to return to the normal mode.

no idea what you mean by this.

btw there's a new `dyslexic` option. if you set `kj` as your main sequence you can also use `jk`.

can you explain a bit more your issue, and i'll find some solutions. thanks.

**Comment by GreenRuslan** [comment]
No problem. I'll try to say it in other words. I'll describe to you my use case and preconditions and I think you will get the idea after that. 

I use two languages. English and Ukrainian. To enter normal mode I use sequence 'kj'. So when I switch to Ukrainian all vim motions work as expected despite it doesn't use Latin alphabet, but when I try to use sequence 'kj' to enter the normal mode, in Ukrainian it is "ло" and it doesn't enter normal mode. 
How I usually solve this problem in different text editors that support vim mode and vimrc (like obsidian). I just add one more sequence to enter normal mode. Like this 'imap ло <Esc>'.

So it would be great if there would be the same option in your app, because I'm pretty sure that many people use several languages for communication and it will be very helpful. 

Thank you for your quick response.

**Reply by godbout** [reply]
ok, so i think there's two things here:

1. if all the Vim moves work well when you're using Ukrainian, maybe i can check if it should also work when entering Normal Mode. maybe it's a bug
2. when it comes to remapping etc. on macOS, there are very good tools made especially for that. even open source, free. so to remap whatever you want you'd better use [Karabiner-Elements](https://github.com/pqrs-org/Karabiner-Elements), or [Hammerspoon](https://github.com/Hammerspoon/hammerspoon). probably KE is preferred as 1) it's a virtual keyboard, so it takes precedence over everything 2) it's its specialty. then you can remap according to whatever you want: the app you're in, some state (for example kV sends Distributed Notifications that you can catch, and you can even remap differently according to different modes you're in. see README).

so yeah, not gonna add multiple sequences sorry, and if there's native remapping, it wayyyyyyyyyyy down the line. more important stuff to do first, even more when there's already tools, and specialized that solve the problem greatly on macOS.

**Reply by godbout** [reply]
can you let me know two things:

1. have you tried Karabiner-Elements and/or Hammerspoon? does it solve your issue
2. what do you expect once you're in Normal Mode. i've tried Ukrainian and the `jk` keys on my physical keyboard go down and up, but they're still recognized as `ло`. do you expect that?
